Dalamud/Plugin/Services/ITargetManager.cs

```diff
@@ -0,0 +1,124 @@
+    /// Gets or sets the focus target.
+    /// </summary>
+    public GameObject? FocusTarget { get; set; }
+    
+    /// <summary>
+    /// Gets or sets the previous target.
+    /// </summary>
+    public GameObject? PreviousTarget { get; set; }
+    
+    /// <summary>
+    /// Gets or sets the soft target.
+    /// </summary>
+    public GameObject? SoftTarget { get; set; }
+
+    /// <summary>
+    /// Sets the current target.
+    /// </summary>
+    /// <param name="gameObject">GameObject to target.</param>
+    public void SetTarget(GameObject? gameObject);
+
+    /// <summary>
+    /// Sets the mouseover target.
+    /// </summary>
+    /// <param name="gameObject">GameObject to target.</param>
+    public void SetMouseOverTarget(GameObject? gameObject);
+
+    /// <summary>
+    /// Sets the focus target.
+    /// </summary>
+    /// <param name="gameObject">GameObject to target.</param>
+    public void SetFocusTarget(GameObject? gameObject);
+
+    /// <summary>
+    /// Sets the previous target.
+    /// </summary>
+    /// <param name="gameObject">GameObject to target.</param>
+    public void SetPreviousTarget(GameObject? gameObject);
+
+    /// <summary>
+    /// Sets the soft target.
+    /// </summary>
+    /// <param name="gameObject">GameObject to target.</param>
+    public void SetSoftTarget(GameObject? gameObject);
+    
+    /// <summary>
+    /// Sets the current target.
+    /// </summary>
+    /// <param name="gameObjectAddress">GameObject (address) to target.</param>
+    public void SetTarget(nint gameObjectAddress);
+
+    /// <summary>
+    /// Sets the mouseover target.
+    /// </summary>
+    /// <param name="gameObjectAddress">GameObject (address) to target.</param>
+    public void SetMouseOverTarget(nint gameObjectAddress);
+
+    /// <summary>
+    /// Sets the focus target.
+    /// </summary>
+    /// <param name="gameObjectAddress">GameObject (address) to target.</param>
+    public void SetFocusTarget(nint gameObjectAddress);
+
+    /// <summary>
+    /// Sets the previous target.
+    /// </summary>
+    /// <param name="gameObjectAddress">GameObject (address) to target.</param>
+    public void SetPreviousTarget(nint gameObjectAddress);
+
+    /// <summary>
+    /// Sets the soft target.
+    /// </summary>
+    /// <param name="gameObjectAddress">GameObject (address) to target.</param>
+    public void SetSoftTarget(nint gameObjectAddress);
+
+    /// <summary>
+    /// Clears the current target.
+    /// </summary>
+    public void ClearTarget();
+
+    /// <summary>
+    /// Clears the mouseover target.
+    /// </summary>
+    public void ClearMouseOverTarget();
+
+    /// <summary>
+    /// Clears the focus target.
+    /// </summary>
+    public void ClearFocusTarget();
+
+    /// <summary>
+    /// Clears the previous target.
+    /// </summary>
+    public void ClearPreviousTarget();
+
+    /// <summary>
+    /// Clears the soft target.
+    /// </summary>
+    public void ClearSoftTarget();
```

Comment:
Given all these are basically explicitly defined setters, are they explicitly needed? Can we coalesce the API down to just using the property setter?

(not a question that needs to be answered by the PR author proper - just generally floating it.)

Reply:
There is merit to at least some of the functions being visible, I don't know how else you would allow this interaction:
![image](https://github.com/goatcorp/Dalamud/assets/9083275/bb57b47d-4d72-4b53-a0f4-4fa6be9903e7)

Reply:
I _think_ setting it to just `null` would work?

Reply:
Ah! Yup, exactly right you are. In that case, I'm very much in favor of removing the duplicates from the interface, and leaving them in the implementation.